### PR TITLE
FDG-2439 Org Queue Changes behind interfaces

### PR DESCRIPTION
**Pull request by bhatiakartik**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/Driver.java [hunk 1]
```diff
@@ -2186,6 +2195,43 @@ private TaskRunner launchTask(Task<? extends Serializable> tsk, String queryId,
     return tskRun;
   }
 
+  private void enforceQueue(HiveConf conf) {
+    String queueEnforcerClassName = conf.getVar(HiveConf.ConfVars.QUEUE_ENFORCER_CLASS);
+    QueueEnforcer queueEnforcer = null;
```
Comment (amrk7s) [inline ql/src/java/org/apache/hadoop/hive/ql/Driver.java]: We should have this initialized at init or something and reuse the same QueueEnforcer object.

File: ql/src/java/org/apache/hadoop/hive/ql/Driver.java [hunk 2]
```diff
@@ -2186,6 +2195,43 @@ private TaskRunner launchTask(Task<? extends Serializable> tsk, String queryId,
...
+    try {
+      queueEnforcer = (QueueEnforcer) Class.forName(queueEnforcerClassName)
+          .newInstance();
+    } catch (Throwable e) {
+      throw new RuntimeException("Couldn't enforce queue due to " + e.getMessage());
```
Comment (amrk7s) [inline ql/src/java/org/apache/hadoop/hive/ql/Driver.java:2203]: CLI main thread would exit if we throw runtime exception. Can we throw SQLException instead wherever we are throwing runtime exception?

File: ql/src/java/org/apache/hadoop/hive/ql/Driver.java [hunk 3]
```diff
@@ -2186,6 +2195,43 @@ private TaskRunner launchTask(Task<? extends Serializable> tsk, String queryId,
...
+    if (org.apache.parquet.Strings.isNullOrEmpty(conf.get(Constants.INITIATOR))) {
+      return UserNameWrapper.INSTANCE.getUsername();
+    }
+    return conf.get(Constants.INITIATOR);
+  }
+
+  private void enrichJobName(Task<? extends Serializable> tsk, String queryId, int jobs) {
+    JobNameEnricher jobEnricher = null;
+    try {
+      jobEnricher = (JobNameEnricher) Class.forName(conf.getVar(ConfVars.JOBNAME_ENRICHER_CLASS))
```
Comment (amrk7s) [inline ql/src/java/org/apache/hadoop/hive/ql/Driver.java:2222]: Same. Lets have this initialised elsewhere and reuse. Lets discuss where we can do this.

File: ql/src/java/org/apache/hadoop/hive/ql/propertymodifier/Constants.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.apache.hadoop.hive.ql.propertymodifier;
+
+/**
+ * Created by kartik.bhatia on 14/05/18.
+ */
+public class Constants {
+
+  public static String MAPRED_QUEUE_NAME = "mapreduce.job.queuename";
```
Comment (amrk7s) [inline ql/src/java/org/apache/hadoop/hive/ql/propertymodifier/Constants.java]: These properties should be already available somewhere in hive code or in its mr dependency. Can you check>

File: ql/src/java/org/apache/hadoop/hive/ql/Driver.java (hunk 2 above)
Comment (bhatiakartik) [reply ql/src/java/org/apache/hadoop/hive/ql/Driver.java:2203]: Don't we want CLI main thread to exit in such cases?

**Review by amrk7s — APPROVED** [review]
LGTM

**Pull request merged**